src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -288,13 +288,15 @@ const useAppPolling = (googleProject, workspaceName) => {
   }
   const loadApps = async () => {
     try {
-      const newApps = !!googleProject ?
-        await Ajax(signal).Apps.list(googleProject, { role: 'creator', saturnWorkspaceName: workspaceName }) :
-        []
-      setApps(newApps)
+      const newGoogleApps = !!workspace && isGoogleWorkspace(workspace) ?
+        await Ajax(signal).Apps.list(googleProject, { role: 'creator', saturnWorkspaceName: workspaceName }) : []
+      const newAzureApps = !!workspace && isAzureWorkspace(workspace) ? await Ajax(signal).Apps.getV2AppInfo(workspace.workspace.workspaceId) : []
```

Comment:
TOL I wonder whether `isGoogleWorkspace` could include the `!!workspace` check (presumably it's valid to say that a null workspace is not an Azure workspace)

Reply:
No, that sort of thing led to a lot of confusion-- there was much early code that said "if this isn't a Google workspace (!googleProject), it's an Azure workspace". But the reality is that we have no idea if it's a Google or Azure workspace until we get back the workspace details.

I have found the code to be far less buggy if you explicitly handle the "workspace not loaded" state and don't get into Azure vs. Google until you actually have the workspace details.